src/Compilers/CSharp/Portable/Symbols/Synthesized/GeneratedNameParser.cs

```diff
@@ -172,34 +173,56 @@ internal static bool TryParseAnonymousTypeParameterName(string typeParameterName
             return false;
         }
 
+        private const int sha256LengthBytes = 32;
+        private const int sha256LengthHexChars = sha256LengthBytes * 2;
+
         // A full metadata name for a generic file-local type looks like:
         // <ContainingFile>FN__ClassName`A
-        // where 'N' is the syntax tree ordinal, 'A' is the arity,
+        // where 'N' is the SHA256 checksum of the original file path, 'A' is the arity,
         // and 'ClassName' is the source name of the type.
         //
         // The "unmangled" name of a generic file-local type looks like:
         // <ContainingFile>FN__ClassName
-        private static readonly Regex s_fileTypeOrdinalPattern = new Regex(@"<([a-zA-Z_0-9]*)>F(\d)+__", RegexOptions.Compiled);
+        private static readonly Regex s_fileTypeOrdinalPattern = new Regex($@"<([a-zA-Z_0-9]*)>F([0-9A-F]{{{sha256LengthHexChars}}})__", RegexOptions.Compiled);
```

Comment:
>([a-zA-Z_0-9]*) [](http://example.com/codeflow?start=79&length=15)

Are we certain this is going to cover all possible file names?

Reply:
It is expected to cover all "display file names" that the compiler will put into this position. Characters outside of this set will generally be replaced with `_`.